Xamarin.Forms.Platform.Android/AppCompat/NavigationPageRenderer.cs

```diff
@@ -173,13 +152,52 @@ protected override void Dispose(bool disposing)
 				if (_drawerLayout != null && _drawerListener != null)
 				{
 					_drawerLayout.RemoveDrawerListener(_drawerListener);
+				}
+
+				if (_drawerListener != null)
+				{
+					_drawerListener.Dispose();
 					_drawerListener = null;
 				}
 
-				_drawerToggle = null;
+				if (_drawerToggle != null)
+				{
+					_drawerToggle.Dispose();
+					_drawerToggle = null;
+				}
+
+				if (_backgroundDrawable != null)
+				{
+					_backgroundDrawable.Dispose();
+					_backgroundDrawable = null;
+				}
 
 				Current = null;
 
+				// dispose child renderers at the end to avoid compliations with MasterDetailPage being nested under NavigationPage
```

Comment:
Let's update this comment to something more descriptive, like
"We dispose the child renderers after cleaning up everything related to Drawer Layout in case one of the children is a Master Detail Page (which may dispose of the Drawer Layout)"